apollo-common/src/main/java/com/ctrip/framework/apollo/common/datasource/TitanSettings.java

```diff
@@ -29,35 +29,27 @@
   private String proTitanDbname;
 
   public String getTitanUrl() {
-    Env env = EnvUtils.transformEnv(Foundation.server().getEnvType());
-    switch (env) {
-      case FAT:
-      case FWS:
-        return fatTitanUrl;
-      case UAT:
-        return uatTitanUrl;
-      case TOOLS:
-      case PRO:
-        return proTitanUrl;
-      default:
-        return "";
+    Env env = Env.valueOf(Foundation.server().getEnvType());
+    if (Env.FAT.equals(env) || Env.FWS.equals(env)) {
```

Comment:
I like to use `switch`, but it's seem like that Java doesn't support `switch` an Object.
**expr** in **switch (expr)** required: 'char, byte, short, int, Character, Byte, Short, Integer, String, or an enum'. 
Is there some way to keep `switch`? 
Or using `String` type i.e `switch (Foundation.server().getEnvType())` to implement it?

Reply:
The previous code uses `switch (env)` with no problem...

Reply:
But now there are. Because `env` in `switch (env)` is an `Enum` type before, but now it was changed to an Object, and Java is not supported `switch` an Object...

Reply:
You are right, I didn't notice the `Env` is now a normal class instead of an enum. Which concerns me that it's not backward compatible?

Reply:
Since our purpose is to make it possible and easy to add environments to `apollo-portal`, I have a bold idea:
1. We leave everying in `apollo-core` untouched so that it's backward compatible.
2. We change the `apollo-portal` logic to make it not dependent on the `Env` enum so that we could dynamically add environments.

How do you think?

Reply:
Only change `apollo-portal` is a Great idea. 
It mean that we need to make a new class to represents `Env` ? Or just use `java.lang.String` (with `trim` and `toUpper`) to represent the environment?

Reply:
I think `java.lang.String` (with `trim` and `toUpper`) is OK.

Reply:
There are 111 usages of `com.ctrip.framework.apollo.core.enums.Env` in `apollo-portal`. There will be a large change...

Reply:
Right, let's give more thoughts on this...

Reply:
And I found that after abandon all usages of `com.ctrip.framework.apollo.core.enums.Env` in `apollo-portal`, there are still some problems.
There are 13 usages of `com.ctrip.framework.apollo.core.MetaDomainConsts` in `apollo-portal`, which exists method like `public static String getMetaServerAddress(Env env)` to get meta server address. Should we make a new class to replace `com.ctrip.framework.apollo.core.MetaDomainConsts`?

Or try a new way. Does not use `java.lang.String` to replace `com.ctrip.framework.apollo.core.enums.Env`.
Make a fake class `Env` in `apollo-portal` Instead of use  `java.lang.String`, and write all compatible method. 
For the class like `com.ctrip.framework.apollo.core.MetaDomainConsts`, also make a fake class in `apollo-portal` for it.

Which one is better?

Reply:
I think we may first consider to use the latter one, to write a new `Env` class(maybe using a new name) which is not an enum and rewrite the logic to retrieve meta server addresses in `apollo-portal`.